src/com/flashfla/net/Multiplayer.as

```diff
@@ -523,17 +524,60 @@ package com.flashfla.net
                 server.getRoomList();
         }
 
+        private function currentUserRoomCount():int
+        {
+            const userId:int = server.myUserId;
+            var roomCount:int = 0;
+
+            for each (var tempRoom:Room in rooms)
```

Comment:
Use `room` instead of `tempRoom`, these dont represent temporary room objects, they're the actual rooms.